src/Core/Enums/RuleId.cs

```diff
@@ -28,7 +28,7 @@ public enum RuleId
         ButtonShouldHavePatterns, // check whether button has at least one of three patterns(Invoke,Toggle,ExpandCollapse)
         ButtonInvokeAndTogglePatterns, // Button should not have Invoke and Toggle patterns together.
         ButtonInvokeAndExpandCollapsePatterns, // Button may have Invoke and ExpandCollapse patterns together. (warning)
-        ButtonToggleAndExpandCollapsePatterns, // Button should have have Toggle and ExpandCollapse patterns together.
+        ButtonToggleAndExpandCollapsePatterns, // Button should have Toggle and ExpandCollapse patterns together.
```

Comment:
Should this be **should not**?

```suggestion
        ButtonToggleAndExpandCollapsePatterns, // Button should not have Toggle and ExpandCollapse patterns together.
```

Reply:
"should not" is correct. The user-displayed message for this is "A button must not support both the Toggle and ExpandCollapse patterns"